### PR TITLE
Make BadgeView's accessibilityLabel customizable

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Binary change:

| File | Before | After | Delta |
|------|--------|-------|-------|
| libFluentUI.a | 29,120,456 bytes | 29,121,224 bytes | 768 bytes |

Updated BadgeView's `accessibilityLabel` to allow for a custom label. If a custom accessibility label is not provided, it defaults to `label.text`.

### Verification

Screenshot 2023-02-21 at 1 20 08 PM

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [x] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1592)

### CODE REVIEW

File: ios/FluentUI/Badge Field/BadgeView.swift [hunk 1]
```diff
@@ -521,5 +521,10 @@ open class BadgeView: UIView, TokenizedControlInternal {
 
     // MARK: Accessibility
 
-    open override var accessibilityLabel: String? { get { return label.text } set { } }
+    private var _accessibilityLabel: String?
```
Comment (huwilkes) [inline ios/FluentUI/Badge Field/BadgeView.swift:524]: Should this be on the datasource instead? Seems a bit odd to set the text and the accessibility label in different places.

File: ios/FluentUI/Badge Field/BadgeView.swift [hunk 2]
```diff
@@ -521,5 +521,8 @@ open class BadgeView: UIView, TokenizedControlInternal {
 
     // MARK: Accessibility
 
-    open override var accessibilityLabel: String? { get { return label.text } set { } }
+    open override var accessibilityLabel: String? {
+        get { return dataSource?.accessibilityLabel ?? label.text }
```
Comment (edjamesmsft) [inline ios/FluentUI/Badge Field/BadgeView.swift:525]: Where does this accessibilityLabel come from in the dataSource? I'm trying to find it in code and I can't track it down.

Comment (joannaquu) [reply ios/FluentUI/Badge Field/BadgeView.swift:525]: It's the NSObject's `accessibilityLabel`. Originally I tried to add a new variable in dataSource but the compiler didn't like that.

File: ios/FluentUI/Badge Field/BadgeView.swift [hunk 3]
```diff
@@ -521,5 +521,8 @@ open class BadgeView: UIView, TokenizedControlInternal {
...
+        set { dataSource?.accessibilityLabel = newValue }
```
Comment (huwilkes) [inline ios/FluentUI/Badge Field/BadgeView.swift:526]: I'm not sure we need the set anymore. People should be able to update the accessibility label through the dataSource, so I don't think we need it exposed in two places.

**Review by huwilkes — APPROVED** [review]

**Pull request merged**